src/librustc/ty/maps/mod.rs

```diff
@@ -360,6 +360,9 @@ define_maps! { <'tcx>
     // however, which uses this query as a kind of cache.
     [] fn erase_regions_ty: erase_regions_ty(Ty<'tcx>) -> Ty<'tcx>,
     [] fn fully_normalize_monormophic_ty: normalize_ty_node(Ty<'tcx>) -> Ty<'tcx>,
+
+    [] fn target_features_whitelist: target_features_whitelist_node(CrateNum) -> Rc<Vec<String>>,
+    [] fn target_features_enabled: TargetFeaturesEnabled(DefId) -> Rc<Vec<String>>,
```

Comment:
Could you maybe use `BTreeSet<Symbol>`? Or is that too much of a microoptimization? (I was thinking we could speed up checking for a specific target feature that way)

Reply:
Realistically there will be at most a dozen or so target features, I wouldn't expect a b-tree to beat linear search for such small collections.

Reply:
If we really cared about performance we would use a binary search into a sorted `Vec`.